protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -822,12 +822,31 @@ impl PoolChannelFactory {
         &mut self,
         m: &mut NewTemplate<'static>,
     ) -> Result<HashMap<u32, Mining<'static>>, Error> {
+        PoolChannelFactory::split_outputs(
+            &mut self.pool_coinbase_outputs,
+            m.coinbase_tx_value_remaining,
+        );
+
         let new_job =
             self.job_creator
                 .on_new_template(m, true, self.pool_coinbase_outputs.clone())?;
         self.inner.on_new_extended_mining_job(new_job)
     }
 
+    /// From the spec the coinbase_tx_value_remaining is:
+    /// The value, in satoshis, available for spending in coinbase outputs
+    /// added by the client. Includes both transaction fees and block subsidy.
+    /// Divide this amount by all outputs
+    pub fn split_outputs(pool_coinbase_outputs: &mut Vec<TxOut>, coinbase_tx_value_remaining: u64) {
```

Comment:
I see the output amounts are calculated by dividing integers evenly - what about remainder, if it is not a perfect division?

Reply:
Looks good!